specification/monitor/resource-manager/Microsoft.Insights/stable/2021-09-01/actionGroups_API.json

```diff
@@ -0,0 +1,1120 @@
+    },
+    "TestNotificationResponse": {
+      "description": "The response when test notification succeeded",
+      "type": "object",
+      "properties": {
+        "notificationId": {
+          "type": "string",
+          "description": "The notification id"
+        },
+        "correlationId": {
+          "type": "string",
+          "description": "The correlation id"
+        },
+        "createdTime": {
+          "type": "string",
+          "description": "The created time"
+        }
+      },
+      "required": [
+        "notificationId",
+        "correlationId",
+        "createdTime"
+      ]
+    },
+    "TestNotificationDetailsResponse": {
+      "description": "The details of the test notification results.",
+      "type": "object",
+      "properties": {
+        "Context": {
+          "type": "object",
+          "$ref": "#/definitions/Context"
+        },
+        "State": {
+          "type": "string",
+          "description": "The overall state"
+        },
+        "CompletedTime": {
+          "type": "string",
+          "description": "The completed time"
+        },
+        "CreatedTime": {
+          "type": "string",
+          "description": "The created time"
+        },
+        "CreatedBy": {
+          "type": "string",
+          "description": "The created by"
+        },
+        "ActionDetails": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ActionDetail"
+          },
+          "description": "The list of action detail"
+        }
+      },
+      "required": [
+        "State"
+      ]
+    },
+    "ActionDetail": {
+      "description": "The action detail",
+      "type": "object",
+      "properties": {
+        "MechanismType": {
+          "type": "string",
+          "description": "The mechanism type"
+        },
+        "Name": {
+          "type": "string",
+          "description": "The name of the action"
+        },
+        "Status": {
+          "type": "string",
+          "description": "The status of the action"
+        },
+        "SubState": {
+          "type": "string",
+          "description": "The substatus of the action"
+        },
+        "SendTime": {
+          "type": "string",
+          "description": "The send time"
+        },
+        "Detail": {
+          "type": "string",
+          "description": "The detail"
+        }
+      }
+    },
+    "Context": {
+      "description": "The context info",
+      "type": "object",
+      "properties": {
+        "NotificationSource": {
+          "type": "string",
+          "description": "The source of the notification request"
+        },
+        "ContextType": {
+          "type": "string",
+          "description": "The context id type"
+        }
+      }
+    },
+    "ActionGroupResource": {
+      "description": "An action group resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureResource"
```

Comment:
Recommend refactoring to reference TrackedResource from https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/types.json to ensure you are following the standard resource patterns